rewrite-xml/src/main/java/org/openrewrite/xml/AutoFormat.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright 2020 the original author or authors.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * https://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.openrewrite.xml;
+
+import org.openrewrite.ExecutionContext;
+import org.openrewrite.Tree;
+import org.openrewrite.xml.search.FindIndentXml;
+import org.openrewrite.xml.tree.Xml;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+import static java.util.Arrays.stream;
+import static java.util.stream.IntStream.range;
+
+public class AutoFormat extends XmlProcessor<ExecutionContext> {
+    private final Xml.Tag[] scope;
+
+    public AutoFormat(Xml.Tag... scope) {
+        this.scope = scope;
+        setCursoringOn();
+    }
+
+    private FindIndentXml findIndent;
```

Comment:
I'm entirely not sure about storing findIndent as an instance variable here. This was previously done in Formatter.wholeSourceIndent.

Reply:
Is this a case where we should have an AutoFormat recipe and AutoFormat and IndentXml processors that communicate via ExecutionContext?